single_kernel_mongo/managers/config.py

```diff
@@ -224,59 +261,83 @@ def binding_ips(self) -> list[str]:
             and self.state.substrate == Substrates.VM
             and not self.state.app_peer_data.external_connectivity
         ):
-            return [
-                f"--bind_ip {self.workload.paths.socket_path}",
-                "--filePermissions 0766",
-            ]
-        return ["--bind_ip_all"]
+            return {
+                "net": {
+                    "bindIp": f"{self.workload.paths.socket_path}",
+                    "unixDomainSocket": {
+                        "filePermissions": "0766",
+                    },
+                },
+            }
+        return {"net": {"bindIpAll": True}}
 
     @property
-    def log_options(self) -> list[str]:
+    def log_options(self) -> dict[str, Any]:
         """The arguments for the logging option."""
-        return [
-            "--setParameter processUmask=037",  # Required for log files permissions
-            "--logRotate reopen",
-            "--logappend",
-            f"--logpath={self.workload.paths.log_file}",
-        ]
+        return {
+            "setParameter": {"processUmask": "037"},
+            "systemLog": {
+                "logRotate": "reopen",
+                "logAppend": True,
+                "path": f"{self.workload.paths.log_file}",
+                "destination": "file",
+            },
+        }
 
     @property
-    def audit_options(self) -> list[str]:
+    def audit_options(self) -> dict[str, Any]:
         """The argument for the audit log options."""
-        return [
-            f"--auditDestination={AuditLogConfig.destination}",
-            f"--auditFormat={AuditLogConfig.format}",
-            f"--auditPath={self.workload.paths.audit_file}",
-        ]
+        return {
+            "auditLog": {
+                "destination": AuditLogConfig.destination,
+                "format": AuditLogConfig.format,
+                "path": f"{self.workload.paths.audit_file}",
+            }
+        }
 
     @property
-    def auth_parameter(self) -> list[str]:
+    def auth_parameter(self) -> dict[str, Any]:
         """The auth mode."""
-        cmd = ["--auth"] if self.auth else []
+        cmd = {"security": {"authorization": "enabled"}} if self.auth else {}
         if self.state.tls.internal_enabled and self.state.tls.external_enabled:
-            return cmd + [
-                "--clusterAuthMode=x509",
-                "--tlsAllowInvalidCertificates",
-                f"--tlsClusterCAFile={self.workload.paths.int_ca_file}",
-                f"--tlsClusterFile={self.workload.paths.int_pem_file}",
-            ]
-        return cmd + [
-            "--clusterAuthMode=keyFile",
-            f"--keyFile={self.workload.paths.keyfile}",
-        ]
+            return always_merger.merge(
```

Comment:
I think we can use python standard ways here, either:
1. return {**cmd, {"security": {} .....}}
2. return cmd | {"security": {} .....}